src/library/utils/R/unix/download.file.R

```diff
@@ -33,14 +34,25 @@ download.file <-
 	method <- if(startsWith(url, "file:")) "internal" else "libcurl"
     }
 
+    if (length(names(headers)) != length(headers) || any(names(headers) == ""))
```

Comment:
`|| anyNA(names(headers))` ?

Reply:
Yeah, in fact, I guess the header itself should not be NA, either...

Reply:
Or should we drop NA headers?